### PR TITLE
Introduction materials

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
I'm kicking off introduction materials.

Some parts needs to be finished, but it should contain all necessary steps:
- git setup & common work flow
- environment setup
- some shallow dive into javascript basics
- React.js hello world

Tips for improvements are welcomed.
Tips for crucial articles that shouldn't be missed in resource list are welcomed.

### CODE REVIEW

File: introduction/environment.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+Try to play with it as you wish!
+
+*Optional exercises:*
+
+- Look into source code and break something, see what happened?
+- Create new file `src/HelloButton.js` with the following content 
+and find out how to add this into page:
+ 
+```jsx harmony
+import React, {Component} from 'react';
```
Comment (PavelGavlik) [inline introduction/environment.md]: This looks somewhat broken when rendered.

File: introduction/git_basics.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+and *fork* the repository
+
+2. **Clone your newly created repository:**
+```bash
+git clone <url-of-your-repo>
+```
+
+Copypasting url from browser is enough.
+
+3. **Create new branch**
```
Comment (PavelGavlik) [inline introduction/git_basics.md]: Ordering is broken. Please fix it or consider turning these into headings

File: introduction/git_basics.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+4. **Commit & push your changes**
+
+```bash
+git add *
+git commit -m 'New cool stuff added'
+git push
+```
+
+5. **Create pull request**
```
Comment (PavelGavlik) [inline introduction/git_basics.md]: Alternatively we can show basics with pushing to master and in second step introduce branches and pull requests.

File: introduction/intro_to_js.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+npm install # installs all the required dependencies (see package.json)
+npm test # run tests
+```
+
+As you can see, tests failed. Can you fix the code so they pass?
+
+## Useful resources
+
+[MDN](https://developer.mozilla.org/cs/docs/Web/JavaScript/) - Unofficial JavaScript wiki 
+[You don't know JS](https://github.com/getify/You-Dont-Know-JS) - Superb well of all JavaScript concepts & traps, both for novices & professionals
```
Comment (PavelGavlik) [inline introduction/intro_to_js.md]: These are displayed as one paragraph, consider using a list

File: introduction/motivation.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+Surprisingly, there is even article about it: [How it feels to learn JavaScript in 2016](https://hackernoon.com/how-it-feels-to-learn-javascript-in-2016-d3a717dd577f)
+
+So you might be asking yourself - why learn JavaScript?
+
+## Why JavaScript
+
+**It's everywhere!** not only in your browser:
+  - [Electron](http://electron.atom.io/) - framework for desktop apps, 
+    used already by several companies, e.g. guess what's behind Slack... 
+  - [React Native](https://facebook.github.io/react-native/) for native mobile applications
```
Comment (PavelGavlik) [inline introduction/motivation.md:16]: We can add more examples of using JS, e.g. Facebook app for React Native or Netflix as company that has Node.js backend.

**Review by PavelGavlik — COMMENTED** [review]
Overall it looks good, I found mostly minor formatting issues. Lastly I would recommend renaming files to 0 xxx.md, 1 xxx.md to give some order in which these documents should be read.

File: introduction/environment.md (hunk 1 above)
Comment (Aprillion) [reply introduction/environment.md]: looks like github supports `jsx` without harmony - https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L1953

File: introduction/environment.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+automatically reload your app when something has changed, run tests and much more.
+
+## Setup
+
+Install node.js if you haven't done so:
+
+**macOS:**
+
+```bash
+brew install node@6
```
Comment (Aprillion) [inline introduction/environment.md:19]: why not 7 ?

File: introduction/intro_to_js.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+```
+var doesntlookLikeNumber = "abc123";
+var result = doesntlookLikeNumber * 2;
+console.log(result); // Outputs "NaN" (Not a number)
+console.log(typeof result); // Outputs "number"
+```
+
+Isn't it fun?
+
+### Truthy/Falsy
```
Comment (Aprillion) [inline introduction/intro_to_js.md:46]: we can link to https://dorey.github.io/JavaScript-Equality-Table/

File: introduction/intro_to_js.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+}
+``` 
+
+### Scopes
+
+![Function scope](/assets/img/function_scope.png)
+
+TODO?
+
+### Variables defined with let & const
```
Comment (Aprillion) [inline introduction/intro_to_js.md]: could be `####` (belongs under Scopes)

File: introduction/git_basics.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+    - Go back to your Github repository
+    - Click on "New pull request"
+    - Select your branch in second dropdown you want to merge
+    - You can use "Reviewers" in righ panel to assign someone, e.g. some lectors, to review your changes
+
+*Tip*: Use ssh keys instead of username/password:
+
+- Go to [github.com/settings/keys](https://github.com/settings/keys) or by clicking on your icon in righ top panel > **settings** > **SSH and GPG keys**
```
Comment (PavelVanecek) [inline introduction/git_basics.md]: Typo: _righ top_

File: introduction/intro_to_js.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+ 
+```
+if (!someVariable) { // outputs "Yes" if someVariable is "", 0, NaN, null, undefined or false 
+    console.log("Yes");
+}
+``` 
+
+### Scopes
+
+![Function scope](/assets/img/function_scope.png)
```
Comment (PavelVanecek) [inline introduction/intro_to_js.md:56]: I would add an example of block scope (let, const, & other languages) as opposed to function scope (var). That can often be a source of confusion and this image does not make it clearer.

File: introduction/intro_to_js.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+TODO?
+
+### Variables defined with let & const
+
+Normally, You already know, `var` defines variables in the scope of whole function.
+This doesn't apply to let & const, which are defined for blocks:
+
+```
+var getName = (isSystem) {
```
Comment (PavelVanecek) [inline introduction/intro_to_js.md]: Missing either `=>` or `function`

File: introduction/environment.md (hunk 6 above)
Comment (jaroslav-kubicek) [reply introduction/environment.md:19]: I don't have high trust in marriage between latest node & win platform, but I will think about it.

**Pull request merged**